crates/astria-sequencer/src/service/consensus.rs

```diff
@@ -200,86 +242,148 @@ impl Consensus {
 
     #[instrument(skip(self))]
     async fn commit(&mut self) -> anyhow::Result<response::Commit> {
+        // clear the cache of transaction execution results
+        self.execution_result.clear();
+        self.processed_txs = 0;
+
         let app_hash = self.app.commit(self.storage.clone()).await;
         Ok(response::Commit {
             data: app_hash.0.to_vec().into(),
             ..Default::default()
         })
     }
-}
 
-/// Generates a commitment to the `sequence::Actions` in the block's transactions.
-/// This is required so that a rollup can easily verify that the transactions it
-/// receives are correct (ie. we actually included in a sequencer block, and none
-/// are missing)
-/// It puts this special "commitment" as the first transaction in a block.
-/// When other validators receive the block, they know the first transaction is
-/// supposed to be the commitment, and verifies that is it correct.
-#[instrument]
-fn handle_prepare_proposal(
-    prepare_proposal: request::PrepareProposal,
-) -> response::PrepareProposal {
-    // generate commitment to sequence::Actions and commitment to the chain IDs included in the
-    // sequence::Actions
-    let res = generate_sequence_actions_commitment(prepare_proposal.txs);
-
-    response::PrepareProposal {
-        txs: res.into_transactions(),
+    /// Generates a commitment to the `sequence::Actions` in the block's transactions.
+    /// This is required so that a rollup can easily verify that the transactions it
+    /// receives are correct (ie. we actually included in a sequencer block, and none
+    /// are missing)
+    /// It puts this special "commitment" as the first transaction in a block.
+    /// When other validators receive the block, they know the first transaction is
+    /// supposed to be the commitment, and verifies that is it correct.
+    #[instrument(skip(self))]
+    async fn handle_prepare_proposal(
```

Comment:
I'm going to turn these into methods on `app` actually now that they're stateful. kinda messy to have them here